bin/node/runtime/src/constants.rs

```diff
@@ -59,6 +59,8 @@ pub mod time {
 	// 1 in 4 blocks (on average, not counting collisions) will be primary BABE blocks.
 	pub const PRIMARY_PROBABILITY: (u64, u64) = (1, 4);
 
+	// NOTE: Currently it is not possible to change the epoch duration after the chain has started.
```

Comment:
Why isn't this documented in Babe etc? Here the docs are not that useful IMHO.

Reply:
It's already documented in BABE: https://github.com/paritytech/substrate/blob/master/frame/babe/src/lib.rs#L119-L123

Reply:
It is documented in Babe, but people will "be inspired" by the Substrate config of the pallets. The team (HydraDX) assumed (for good reason IMO) that they could change the config later with a runtime upgrade if necessary.

Reply:
I think that the bigger issue is that even if you read that, the implications of chosing 10 minute epoch are not perfectly clear. 

It would seem that it's completely fine, until you realize that validators need to post imOnline message in this time to keep in the set (which doesn't seem horrible) but they also get denominated? (I think), which is probably worse. And also that if you skip one session, the chain will stall which looks like THE thing you don't want that. I've since read the research paper on that but some sane defaults could be part of the original documentation (which is around 3.3 to 4 hours from what I've read)

Reply:
If you are fine with 10 minutes, why not modify the rest to not remove validators if they are offline for example? Or to not get them denominated? 

You have the freedom to do this.

Reply:
@bkchr We have considered this. There are probably multiple reasons why it is that validators should post imOnline status during an epoch, since election processes and randomness generation are also tied to this I can imagine that it could have security implications as also stated here: https://research.web3.foundation/en/latest/polkadot/block-production/Babe.html (was going through this yesterday). 10 minute no-block stall also looks like a valid problem even though the chain was running smoothly up until now. It was advised by multiple people that we should change this. 

Actually I was more leaned to try to run it, but we made a consensus that coordinating an upgrade now is much better than coordinating the chaos that could potentially happen and trying to fix that once fully running with more validators. As right now, we're in POA stage of BABE where only our nodes are validating in the first era. We have 200+ waiting validators and several thousand participants so it looks like a saner (and possibly quicker) option.